tests/test_treasury.cairo

```diff
@@ -0,0 +1,292 @@
+    let (gov_contract_address, _AMM_contract_address, treasury_contract_address, _) =
+        get_important_addresses();
+    let new_AMM_contract: ContractAddress = '0xnewAMMcontract'.try_into().unwrap();
+
+    prank(
+        CheatTarget::One(treasury_contract_address), gov_contract_address, CheatSpan::TargetCalls(1)
+    );
+    ItreasuryDispatcher { contract_address: treasury_contract_address }
+        .update_AMM_contract(new_AMM_contract);
+
+    let recorded_AMM_addr = ItreasuryDispatcher { contract_address: treasury_contract_address }
+        .get_amm_address();
+    assert(new_AMM_contract == recorded_AMM_addr, 'Error updating AMM address');
+}
+
+#[test]
+#[should_panic(expected: ('Caller is not the owner',))]
+fn test_update_AMM_contract_by_unauthorized_caller() {
+    let (_gov_contract_address, _AMM_contract_address, treasury_contract_address, _) =
+        get_important_addresses();
+    let user2: ContractAddress = '0xUser2'.try_into().unwrap();
+    let new_AMM_contract: ContractAddress = '0xnewAMMcontract'.try_into().unwrap();
+
+    prank(CheatTarget::One(treasury_contract_address), user2, CheatSpan::TargetCalls(1));
+    ItreasuryDispatcher { contract_address: treasury_contract_address }
+        .update_AMM_contract(new_AMM_contract);
+
+    let recorded_AMM_addr = ItreasuryDispatcher { contract_address: treasury_contract_address }
+        .get_amm_address();
+    assert(new_AMM_contract == recorded_AMM_addr, 'unAuthorized AMM update');
+}
+
+#[test]
+#[fork("MAINNET2")]
+fn test_provide_liquidity_to_carm_AMM() {
+    let (gov_contract_address, _AMM_contract_address1, treasury_contract_address, _) =
+        get_important_addresses();
+    let quote_token: ContractAddress =
+        0x53c91253bc9682c04929ca02ed00b3e423f6710d2ee7e0d5ebb06f3ecf368a8
+        .try_into()
+        .unwrap();
+    let base_token: ContractAddress =
+        0x49d36570d4e46f48e99674bd3fcc84644ddd6b96f7c741b1562b82f9e004dc7
+        .try_into()
+        .unwrap();
+    let pooled_token: ContractAddress =
+        0x49d36570d4e46f48e99674bd3fcc84644ddd6b96f7c741b1562b82f9e004dc7
+        .try_into()
+        .unwrap();
+    let newUser: ContractAddress =
+        0x0529420740c3a42dd41f842ac5c03bd0ac4a56780bfd500dfc8488df8d016e2e
+        .try_into()
+        .unwrap();
+
+    let AMM_hash: felt252 = 0x045fb686c8875f31966e7308d71c03e9ae78f9566a61870a2b616dc225dd3313;
+    let mut calldata1 = ArrayTrait::new();
+    gov_contract_address.serialize(ref calldata1);
+    let AMM_contract_address = ContractClass { class_hash: AMM_hash.try_into().unwrap() }
+        .deploy(@calldata1)
+        .unwrap();
+
+    let put_lp_token_hash: felt252 =
+        0x06d15bc862ce48375ec98fea84d76ca67b7ac5978d80c848fa5496108783fbc2;
+    let lp_token_name = 'Carmine ETH/USDC call pool';
+    let lp_token_symbol = 'C-ETH/USDC';
+    let mut calldata = ArrayTrait::new();
+    lp_token_name.serialize(ref calldata);
+    lp_token_symbol.serialize(ref calldata);
```

Comment:
This test should be testing only the deposit of liquidity, instead, it is testing the creation of a new pool.

If the issue is lack of tokens, you can just mint the tokens by pranking the token contract or transferring the tokens from someone who has them by pranking their address (sending transfer as their address) in the fork test.